src/panels/lovelace/mixins/element-click-mixin.js

```diff
@@ -12,8 +13,78 @@ import computeStateName from "../../../common/entity/compute_state_name";
 export default dedupingMixin(
   (superClass) =>
     class extends NavigateMixin(EventsMixin(superClass)) {
-      handleClick(hass, config) {
-        const tapAction = config.tap_action || "more-info";
+      registerMouse(config) {
+        let ripple = null;
+
+        const createRipple = () => {
+          if (ripple === null) {
+            ripple = document.createElement("mwc-ripple");
+            this.shadowRoot.querySelector("div").appendChild(ripple);
```

Comment:
It feels weird that we have to insert divs into the elements. I think that if we set the elements to render `display: inline-block`, it should be able to ripple in the host.

Reply:
Here's the problem: https://github.com/material-components/material-components-web-components/blob/master/packages/ripple/src/ripple-directive.ts#L79
It tries to read and set `.style.position` of its parent node, and shadowRoot doesn't have a style property.

And if the mwc-ripple is outside of shadowRoot all kinds of weird things happen...

Reply:
You should never hit that code as both are in the same shadow root and thus have same root node?

Reply:
Ah, I see that they are passing the holder of the shadow root as interaction node: https://github.com/material-components/material-components-web-components/blob/master/packages/ripple/src/mwc-ripple.ts#L41

So it's not the same. 

What kind of weird things happen ? Did you try setting the host to `display: inline-block` ?

Reply:
Actually, we could override interactionNode on the ripple to be the first child in the shadowRoot. That means the root nodes will be the same

Reply:
Complicated 😄

Reply:
If I do
```javascript
this.shadowroot.appendChild(ripple);
console.log(ripple.parentNode);
```
The console shows
```
#shadowRoot (open)
```
followed by

```
Uncaught (in promise) TypeError: Cannot read property 'position' of undefined
at rippleNode (ripple-directive.ts:79)
```

If I do
```
this.appendChild(ripple);
```

The entity jumps around and lands somewhere entirely different as soon as I click it. Adding `display: inline-block` to `:host` doesn't fix it.

Reply:
> Complicated

Tell me about it!

Reply:
Overriding interactionNode with `this.shadowRoot.childNodes[2]` seems to work... but that certainly doesn't seem robust.

Child node 0 is `<style>` and child node 1 is `#text`, which I don't even know what it is... And that's just for the *current* elements.